### DISCUSSION TITLE
Dynamically import pdf.js v4.0.189 in Angular 17 project using esbuild

### DISCUSSION THREAD

**Discussion by benbro** [mozilla/pdf.js#17288] [Q&A]
I'm trying to build and dynamically import pdf.js v4.0.189 in an Angular 17 project using esbuild.
It used to work with pdf.js v3.11.174 and webpack.

When building the project I'm getting an error:
```
⠙ Building...✘ [ERROR] Top-level await is not available in the configured target environment ("chrome119.0", "edge119.0", "firefox115.0", "ios17.0", "safari17.0" + 5 overrides)

    node_modules/pdfjs-dist/build/pdf.mjs:16869:44:
      16869 │ ...ck_exports__ = globalThis.pdfjsLib = await __webpack_exports__;
```

I've tried to wrap line 16837 in /node_modules/pdfjs-dist/build/pdf.mjs with an invoked function:
```javascript
/******/ (async () => {__webpack_exports__ = globalThis.pdfjsLib = await __webpack_exports__;})();
```

Now I'm able to build but when trying to dynamically import the module pdfjsLib.GlobalWorkerOptions and all other attributes are undefined:
```javascript
import('pdfjs-dist/build/pdf.mjs').then(pdfjsLib => {
    console.log(pdfjsLib.GlobalWorkerOptions);
});
```
Is it ok to manually wrap the line with ```(async () => {})();```?
Is there a way to dynamically import pdfjsLib in esbuild project and set pdfjsLib.GlobalWorkerOptions.workerSrc?

**Comment by btrewern** [comment]
It's an esbuild issue (or pdf.js issue depending on your point of view), see here: https://github.com/mozilla/pdf.js/issues/17245.
